### PR TITLE
Document Timer autostart in tool scripts

### LINKED ISSUE #63188: [Godot4.0] Timer not auto start when tool mode
### Godot version

v4.0.alpha12.official [2c11e6d9e]

### System information

Windows 10 64bit

### Issue description

Code like this:
```
@tool
extends Node2D

func _init():
	var timer = Timer.new()
	timer.wait_time = 1
	timer.autostart = true
	add_child(timer)
	timer.timeout.connect(update)

func _ready():
	pass


func update():
	prints(Time.get_ticks_msec())
```
But the timer not auto start, I must start it like this:

### Steps to reproduce

Please run the reprocution project

### Minimal reproduction project

[TimerToolTest.zip](https://github.com/godotengine/godot/files/9138870/TimerToolTest.zip)

### PR DESCRIPTION
**Pull request by arkology**
Closes https://github.com/godotengine/godot/issues/63188

### CODE REVIEW

File: doc/classes/Timer.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
 	<members>
 		<member name="autostart" type="bool" setter="set_autostart" getter="has_autostart" default="false">
 			If [code]true[/code], the timer will start immediately when it enters the scene tree.
-			[b]Note:[/b] After the timer enters the tree, this property is automatically set to [code]false[/code].
+			[b]Note:[/b] After the timer enters the tree, this property is automatically set to [code]false[/code]. If the timer will be the part of the scene currently opened in the editor (for example, adding and running timer from [code]@tool[/code] script), this property will not set to [code]false[/code] and the timer will not start immediately when it enters the scene tree.
```
Comment (Mickeon) [inline doc/classes/Timer.xml:45]: Can't it just be...?
```suggestion
			[b]Note:[/b] After the timer enters the tree, this property is automatically set to [code]false[/code].
			[b]Note:[/b] This property does nothing when the timer is running in the editor.
```

**Review by Mickeon — COMMENTED** [review]
Somewhat verbose

File: doc/classes/Timer.xml (hunk 1 above)
Comment (arkology) [reply doc/classes/Timer.xml]: Sure!
This variant will look like my first version of changes, which I rejected as "not fully descriptive" :D

**Review by Mickeon — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**